src/java.base/share/classes/java/lang/StableValue.java

```diff
@@ -0,0 +1,664 @@
+ * are blocked until the stable value becomes set.
+ * <p>
+ * The at-most-once write operation on a stable value that succeeds
+ * (e.g. {@linkplain #trySet(Object) trySet()})
+ * {@linkplain java.util.concurrent##MemoryVisibility <em>happens-before</em>}
+ * any subsequent read operation (e.g. {@linkplain #orElseThrow()}).
+ * <p>
+ * The method {@linkplain StableValue#orElseSet(Supplier) orElseSet()} guarantees that
+ * the provided {@linkplain Supplier} is invoked successfully at most once even under
+ * race. Since stable functions and stable collections are built on top of
+ * {@linkplain StableValue#orElseSet(Supplier) orElseSet()} they too are
+ * thread safe and guarantee at-most-once-per-input invocation.
+ *
+ * @implSpec Implementing classes of {@linkplain StableValue} are free to synchronize on
+ *           {@code this} and consequently, care should be taken whenever
+ *           (directly or indirectly) synchronizing on a {@code StableValue}. Failure to
+ *           do this may lead to deadlock. Stable functions and collections on the
+ *           other hand are guaranteed <em>not to synchronize</em> on {@code this}.
+ *           Except for a {@code StableValue}'s content itself, an {@linkplain #orElse(Object) orElse(other)}
+ *           parameter, and an {@linkplain #equals(Object) equals(obj)} parameter; all
+ *           method parameters must be <em>non-null</em> or a {@link NullPointerException}
+ *           will be thrown.
+ *
+ * @implNote A {@linkplain StableValue} is mainly intended to be a non-public field in
+ *           a class and is usually neither exposed directly via accessors nor passed as
+ *           a method parameter.
+ *           As objects can be set via stable values but never removed, this can be a source
+ *           of unintended memory leaks. A stable value's content is
+ *           {@linkplain java.lang.ref##reachability strongly reachable}. Clients are
+ *           advised that {@linkplain java.lang.ref##reachability reachable} stable values
+ *           will hold their set content perpetually.
+ *           A {@linkplain StableValue} that has a type parameter {@code T} that is an array
+ *           type (of arbitrary rank) will only allow the JVM to treat the <em>array reference</em>
+ *           as a stable value but <em>not its components</em>. Clients can instead use
+ *           {@linkplain #list(int, IntFunction) a stable list} of arbitrary depth, which
+ *           provides stable components. More generally, a stable value can hold other
+ *           stable values of arbitrary depth and still provide transitive constantness.
+ *
+ * @param <T> type of the content
+ *
+ * @since 25
+ */
+@PreviewFeature(feature = PreviewFeature.Feature.STABLE_VALUES)
+public sealed interface StableValue<T>
+        permits StableValueImpl {
+
+    // Principal methods
+
+    /**
+     * {@return {@code true} if the content was set to the provided {@code value},
```

Comment:
I find this description a bit confusing -- as written it almost looks like a predicate -- e.g. test if this stable value is set with the value I'm providing. But the important thing here is the side-effect -- so I feel the javadoc for this method should be much less about what this method returns, and much more about what the method actually does.

Reply:
I also note that, so far, we have always referred to the thing contained in a stable value as "content" (bonus points for that -- as that's also the terminology we settled on in the JEP!). But here (and in other methods) we fall back to `value` -- which is generic, and also a bit confusing with respect to `StableValue` itself.